### ISSUE TITLE
`MeshDataTool.get_vertex_normal` returns incorrect normals

### ISSUE THREAD

**Issue by per2plex** [godotengine/godot#65619]
### Godot version

4.0.alpha16.official

### System information

Windows 10

### Issue description

Since v4.0.alpha15.official `MeshDataTool.get_vertex_normal` doesn't return correct normals for the mesh.
This happens with both imported meshes (left) and primitive meshes (right), see screenshot.

Might be related to the octahedral normal compression:
https://github.com/godotengine/godot/pull/60309/

**Alpha 14:**
![alpha14](https://user-images.githubusercontent.com/1639710/189486495-b2aae1e4-d5d1-495e-aa51-7fe40c7d2a54.png)

**Alpha 15 & 16:**
![alpha16](https://user-images.githubusercontent.com/1639710/189486499-b57fb35f-1a78-43c1-85a8-9f0d52e3bc93.png)

### Steps to reproduce

Run the attached project in either alpha 15 or alpha 16 and notice that the drawn mesh normals are not correct.

### Minimal reproduction project

[meshtool-normal-issue.zip](https://github.com/godotengine/godot/files/9540517/meshtool-normal-issue.zip)

**Comment by maniacalipsis** [comment]
I didn't used MeshDataTool.get_vertex_normal, but since alpha15 I noticed that shadows appears broken. When I switched display mode to Normal Buffer, it looks like all normals was shuffled (whereas in the import preview meshes was displayed ok).
Moreover, in the scenes with the custom environment all meshes was rendered blury with a huge halo around its contours (both imported with materials and primitives without materials).
The problem was solved by deleting .godot/imported and .godot/shader_cashe from the project.
